src/bun.js/bindings/bindings.cpp

```diff
@@ -2497,9 +2497,13 @@ JSC__JSValue JSC__JSValue__keys(JSC__JSGlobalObject* globalObject, JSC__JSValue
 bool JSC__JSValue__hasOwnProperty(JSC__JSValue jsValue, JSC__JSGlobalObject* globalObject, ZigString key)
 {
     JSC::VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
 
     JSC::JSValue value = JSC::JSValue::decode(jsValue);
-    return value.toObject(globalObject)->hasOwnProperty(globalObject, JSC::PropertyName(JSC::Identifier::fromString(vm, Zig::toString(key))));
+    JSObject* obj = value.toObject(globalObject);
```

Comment:
This potentially coerces, it should be `!isObject` and return false

Reply:
```suggestion
     if (!value.isObject()) return false;
    JSObject* obj = value.getObject();
```

Reply:
we need to coerce because given an array or string the indexes should become properties